core-bundle/src/EventListener/SearchIndexListener.php

```diff
@@ -129,6 +141,18 @@ private function needsDelete(Response $response, Document $document): bool
             return true;
         }
 
+        // searchIndexer setting has priority over robots tag
+        $pageJsonLds = $document->extractJsonLdScripts('https://schema.contao.org/', 'Page');
+        $pageSearchIndexer = $pageJsonLds[0]['searchIndexer'] ?? null;
+
+        if ('always_index' === $pageSearchIndexer) {
+            return false;
+        }
+
+        if ('never_index' === $pageSearchIndexer) {
+            return true;
+        }
```

Comment:
This should be moved above the `X-Robots-Tag` check as responses with `X-Robots-Tag: noindex` and `"searchIndexer": "always_index",` should not be deleted I think.

Reply:
See [above](https://github.com/contao/contao/pull/8252#discussion_r2169889086).